ansible-ipi-install/roles/bootstrap/tasks/10_load_inv.yml

```diff
@@ -172,14 +166,31 @@
     msg: Required number of nodes are not available for workers
   when: worker_mgmts | length < worker_count | int
 
-# Needed in case where worker_mgmts is empty list, in that case worker_fdns fact is not set and there is a failure in set-deployment-facts role
-- name: Set fact for empty worker_fqdns
+# Needed in case where lists are constructed in a loop that might never be run 
+# if with_items list is empty
+- name: Set fact for empty zero worker
```

Comment:
Do we need the other vars also initialized to empty list here? I thought `worker_vendors` and `worker_types` are sufficient?

Reply:
Most of the other vars also can also fail to be initialized if the loop body that appends to them is never executed.   This happens when the machine type is not recognized, for example.   I think it's cleaner code to initialize them up front to the empty list and then just append to them as needed.    But I could code it with less changes if we assume that the machine types are always defined.

Reply:
yes, `worker_vendors` and `worker_types` are REQUIRED, other are good to have.